speccurses.py

```diff
@@ -24,90 +26,146 @@
+
+def display_run(stdscr, runinfo, y=STARTY + 2):
+    """
+    Displays a single run within a config file
+    :param stdscr: The current screen
+    :param runinfo: The 'spec_run' to display
+    :param y: The current y value
+    :return: The new y value, as well as the max length of any property in the current run
+    """
     mem = [attr for attr in dir(runinfo) if not callable(getattr(runinfo, attr)) and not attr.startswith("__")]
     valueoffset = len(max(mem, key=len)) + 2
     for name in mem:
-        if(name == 'properties'):
+        if (name == 'properties'):
             stdscr.addstr(y, xoffset, name)
         else:
             stdscr.addstr(y, xoffset, "{}:".format(name))
             stdscr.addstr(y, xoffset + valueoffset, str(getattr(runinfo, name)))
         y += 1
     return y, valueoffset
 
-#Allows user to select from a list of valid options
+
 def select_from(stdscr, x, y, value, list):
+    """
+    Allows user to select from a list of valid options
+    :param stdscr: The current screen
+    :param x: The start x position to begin printing
+    :param y: The start y position to begin pritning
+    :param value: The current value chosen
+    :param list: A list of values to choose from
+    :return: A value within :param list
+    """
     k = 0
     pad = curses.newpad(1, 100)
     height, width = stdscr.getmaxyx()
     idx = list.index(value)
     while (k != ENTER and k != ord('q')):
         pad.clear()
         draw_status_bar(stdscr, "Press 'q' to exit and 'UP' or 'DOWN' to select a value")
-        if(k == curses.KEY_UP and idx > 0):
+        if (k == curses.KEY_UP and idx > 0):
             idx -= 1
-        elif(k == curses.KEY_DOWN and idx < len(list) - 1):
+        elif (k == curses.KEY_DOWN and idx < len(list) - 1):
             idx += 1
         value = list[idx]
-        pad.addstr(0,0, str(value))
+        pad.addstr(0, 0, str(value))
         stdscr.move(y, x + len(str(value)))
         pad.refresh(0, 0, y, x, y, width - x)
         k = stdscr.getch()
     return value
 
-#Allows user to input text and returns the result
+
 def input_text(stdscr, x, y, value, validator):
+    """
+     Allows user to input text and returns the result
+    :param stdscr: The current screen
+    :param x: The start x position to begin printing
+    :param y: The start y position to begin pritning
+    :param value: The current value chosen
+    :param validator: A validator to ensure only proper keys are entered.
+    :return: A validated value
+    """
     k = 0
     value = list(str(value))
     idx = len(value)
     height, width = stdscr.getmaxyx()
-    pad = curses.newpad(1, width - xoffset)
+    xmax = width * 2
+    pad = curses.newpad(1, xmax)
+    cursorx =  min(x + len(value), width - 1)
 
     while (k != ENTER):
         pad.clear()
-        if(k == BACK and idx > 0):
-            if(idx < len(value)):
-                value = value[:idx-1] + value[idx:]
+        if (k == BACK and idx > 0):
```

Comment:
Doesn't `curses` for python provide key values like how `BACK` is used? We should use that instead if so.